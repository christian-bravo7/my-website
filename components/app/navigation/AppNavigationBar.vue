<template>
  <AppNavigationBarBase>
    <template #left>
      <div class="hidden md:flex">
        <AppLogo />
      </div>
    </template>
    <template #center>
      <div class="hidden | w-full | md:flex items-center justify-end col-span-3">
        <div class="flex content-center | mr-6">
          <AppNavigationBarLink
            :to="localePath({ name: 'index' })"
          >
            {{ $t('navigation.home-label') }}
          </AppNavigationBarLink>
          <AppNavigationBarLink
            :to="localePath({ name: 'me' })"
          >
            {{ $t('navigation.about-me-label') }}
          </AppNavigationBarLink>
        </div>
      </div>
    </template>
    <template #right>
      <div class="hidden | md:flex items-center justify-end col-span-1">
        <div class="mr-4">
          <LanguagePicker />
        </div>
        <div class="flex justify-center items-center">
          <DarkModeSwitch />
        </div>
      </div>
      <div class="flex md:hidden justify-end items-center">
        <AppBurgerButton
          :is-active="isSidebarActive"
          @click="toggleSidebar"
        />
      </div>
    </template>
  </AppNavigationBarBase>
</template>

<script lang="ts">
import { Vue, Component, Emit, Prop } from 'nuxt-property-decorator';

@Component
export default class AppNavigationBar extends Vue {
  @Prop({ type: Boolean, default: false })
  readonly isSidebarActive!: boolean;

  @Emit('openSidebar')
  openSidebar (): void {}

  @Emit('toggleSidebar')
  toggleSidebar (): void {}
}
</script>
