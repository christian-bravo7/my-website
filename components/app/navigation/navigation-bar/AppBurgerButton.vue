<template>
  <button
    class="w-12 h-12 | p-3 | rounded-full | bg-gray-100 | app-burger-button"
    :class="{ 'z-50': isActive }"
    @click="onClick"
    @keydown.enter="onClick"
  >
    <div
      class="flex flex-col items-end justify-center | relative | w-full h-full | app-burger-button__wrapper"
      :class="{ 'app-burger-button__wrapper--active': isActive }"
    >
      <span class="absolute | block | w-9/12 | bg-black" />
      <span class="absolute | block | w-full | bg-black" />
      <span class="absolute | block | w-7/12 | bg-black" />
    </div>
  </button>
</template>

<script lang="ts">
import { Component, Emit, Vue } from 'nuxt-property-decorator';

@Component
export default class AppBurgerButton extends Vue {
  isActive: boolean = false;

  @Emit('click')
  onClick (): void {
    this.isActive = !this.isActive;
  }
}
</script>

<style lang="scss" scoped>
.app-burger-button {
  &__wrapper {
    span {
      height: 2px;
      transition-duration: 300ms;
      transition-property: transform, opacity;
    }

    span:nth-child(1) {
      transform: translateY(-6px);
    }

    span:nth-child(2) {
      transform: translateY(0);
    }

    span:nth-child(3) {
      transform: translateY(6px);
    }

    &--active {
      span:nth-child(1) {
        @apply w-full;

        transform: rotate(45deg);
      }

      span:nth-child(2) {
        opacity: 0;
      }

      span:nth-child(3) {
        @apply w-full;

        transform: rotate(-45deg);
      }
    }
  }
}
</style>
