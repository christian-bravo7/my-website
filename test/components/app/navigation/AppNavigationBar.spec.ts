import { createLocalVue, mount } from '@vue/test-utils';

import AppNavigationBar from '@/components/app/navigation/AppNavigationBar.vue';
import AppBurgerButton from '@/components/app/navigation/AppBurgerButton.vue';

import setupI18n from '@/test/config/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('AppNavigationBar component', () => {
  it('should render correctly', () => {
    const wrapper = mount(AppNavigationBar, {
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should emit onSidebarToggle event when AppBurgerButton is clicked', async () => {
    const wrapper = mount(AppNavigationBar, {
      ...componentConfig,
    });

    const button = wrapper.findComponent(AppBurgerButton);

    await button.trigger('click');

    expect(wrapper.emitted('onSidebarToggle')).toBeTruthy();
  });
});
