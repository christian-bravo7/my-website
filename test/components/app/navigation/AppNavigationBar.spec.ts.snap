// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigationBar component should render correctly 1`] = `
<nav class="fixed top-0 left-0 right-0 z-20 | p-4 md:py-3 md:px-8 | bg-transparent md:bg-gray-50 md:dark:bg-blue-800 | transition-all duration-150 transform transition-transform duration-300 -translate-y-0">
  <div class="navbar__content">
    <div class="flex items-center">
      <div class="hidden md:flex"><a class="flex col-span-1 | w-8 h-8 | text-black dark:text-white"><svg class="w-full h-full | primary-text"></svg></a></div>
    </div>
    <div class="flex items-center">
      <div class="hidden | w-full | md:flex items-center justify-end col-span-3">
        <div class="flex content-center | mr-6"><a class="mx-6 | text-center text-sm | text-pink-500 dark:text-blue-400" to="[object Object]">
            Home
          </a> <a class="mx-6 | text-center text-sm | text-pink-500 dark:text-blue-400" to="[object Object]">
            About me
          </a></div>
      </div>
    </div>
    <div class="flex items-center">
      <div class="hidden | md:flex items-center justify-end col-span-1">
        <div class="mr-4">
          <div class="inline-block">
            <div class="flex items-center">
              <div class="relative | select-none | app-select">
                <div tabindex="0" class="flex items-center | py-1 pl-4 pr-1 | text-sm | text-gray-900 dark:text-gray-50 bg-gray-100 dark:bg-blue-900 | rounded-lg | cursor-pointer | app-select__button"><span class="flex items-center"><i class="material-icons text-sm mr-2">
            translate
          </i>
          
        </span> <i class="material-icons ml-1 | transition-transform duration-300">
                    keyboard_arrow_down
                  </i></div>
                <div tabindex="0" class="absolute right-0 | p-1 | bg-gray-50 dark:bg-blue-900 | rounded-lg shadow-lg focus:outline-none | app-select__options-list" style="display: none;"></div>
              </div>
            </div>
          </div>
        </div>
        <div class="flex justify-center items-center">
          <div class="relative flex select-none"><input id="dark-mode-switch" type="checkbox" class="hidden"> <label for="dark-mode-switch" class="relative | flex justify-center items-center | w-8 h-8 | p-2 | text-yellow-500 dark:text-gray-50 bg-gray-100 dark:bg-blue-900 | cursor-pointer rounded-lg"><span mode="out-in"><i class="absolute | text-xl | material-icons dark-mode-switch__icon">
        mode_night
      </i></span></label></div>
        </div>
      </div>
      <div class="flex md:hidden justify-end items-center"><button class="w-12 h-12 | p-3 | rounded-full | bg-gray-100 dark:bg-blue-900 | app-burger-button">
          <div class="flex flex-col items-end justify-center | relative | w-full h-full | app-burger-button__wrapper"><span class="absolute | block | w-9/12 | bg-black dark:bg-gray-100"></span> <span class="absolute | block | w-full | bg-black dark:bg-gray-100"></span> <span class="absolute | block | w-7/12 | bg-black dark:bg-gray-100"></span></div>
        </button></div>
    </div>
  </div>
</nav>
`;
